utils/das_data_sorter.py

```diff
@@ -1,76 +1,170 @@
+        self.power = self.average_data(data["power"])
+        self.reed_velocity = self.average_data(data["reed_velocity"])
+        self.reed_distance = self.average_data(data["reed_distance"])
+
+        self.print_output_file()
+
+    def convert_time(self, milliseconds:D) -> D:
+        '''Returns the conversion of the time data points from milliseconds to the specified time unit, depending on the 
+        argument passed by args.unit.
+        
+        args.unit accepts only seconds/s and minutes/m as arguments. 
+        '''
+        if args.unit == "seconds" or args.unit == "s":
+            milliseconds = milliseconds/1000
+        elif args.unit == "minutes" or args.unit == "m":
+            milliseconds = milliseconds/1000/60
+        else:
+            raise ValueError("Argument only accepts either seconds/s or minutes/m")
+        
+        # sets the groups of indexes to use for averaging in future
+        self.indexes = self.group_index(milliseconds)
+
+        return range(1,len(self.indexes)+1)
+
+    def group_index(self, time:D) -> D:
+        '''Returns a universal array of arrays of indexes, based on the specified time unit. Each array represents the indexes 
+        of the data points within the same time interval. (eg. 1123ms and 1748ms are in the same time interval for seconds, 
+        but 2453ms isn't)
+
+        When given a valid value for args.unit, function will group the indexes within the same time interval in an array.
+        Once all the indexes of a time interval has been accounted for, the array of indexes will be pushed into a
+        universal array so that each element represents the data points of one singular time interval.
+        '''
+        universal_array = []
+        index_array = []
+        previous_time = 0
+
+        for i in range(len(time)):
+            if time[i] > previous_time:
+                # if true, then time at time[i] is the next second/minute
+                if previous_time != 0:
+                    # appends index array into universal array, but ignores the first iteration
+                    universal_array.append(index_array)
+
+                previous_time = ceil(time[i])
+                index_array = [] # recreates new array if previous time has changed
+            index_array.append(i) 
+
+        # pushes the final array when for loop has completed
+        universal_array.append(index_array)
+
+        return universal_array
+
+    def mean(self, data_array:list) -> int:
+        '''Finds the average of a given set of numbers. 
+
+        Disregards None and zero in calculation. Otherwise, if input is invalid, then raises error.
+        '''
+        total = 0
+        length = len(data_array)
+        for number in data_array:
+            if number == 0 or pd.isna(number):
+                # None and zeroes are ignored
+                length -= 1
+                continue
+            elif number > 0 or number < 0:
```

Comment:
Unless you've thought of some special case I haven't thought of, if `number == 0` was false, then `number > 0 or number < 0` will always be true. So, this case in the if statement will always run, and the `else` section below will never be reached. You should be able to replace the whole thing with `total += number`

Reply:
I had a problem earlier where, in the CSV file, if the cell was empty, it would say that I was trying to perform addition with an integer and a string. (Shown below is an outdated screenshot)
![image](https://user-images.githubusercontent.com/58575113/82114274-15d38780-979f-11ea-98cc-940bcf130ef2.png)

So I thought that an empty cell in Python would read as "". However, after some debugging, an empty cell's data type is `float`, so I'm just all-round confused. Hence why `pd.isna(number)` is there, which is Pandas's way of handling empty cells. I also kept the else code block there for future debugging. 

(Shown below is the error message when elif statement has `number == ""`)
![image](https://user-images.githubusercontent.com/58575113/82113798-128acc80-979c-11ea-9331-14706e785bfe.png)